engine/dlib/src/dlib/hash.cpp

```diff
@@ -668,7 +668,7 @@ DM_DLLEXPORT const char* dmHashReverseSafe64(uint64_t hash)
     const char* s = (const char*)dmHashReverse64(hash, 0);
     if (s == 0)
     {
-        char tmp[64];
```

Comment:
I'm afraid I missed this in the code review:

* Returning stack allocated variables is not good, as they may be invalid when receiving it on the other end.

* The function is not thread safe now, since two different threads will write to the same memory buffer.

I don't have a good way to deal with it as the current api is.
I would introduce a new function for your use case, where to store the variable:

`const char* dmHashReverseSafe64(hash, buffer, buffersize);`